### PR TITLE
cdc: add compatibility notes for sort-dir

### PR DESCRIPTION
**Pull request by liuzix**
### What is changed, added or deleted? (Required)
- Added compatibility notes for `sort-dir` and `data-dir` in TiCDC

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.1 (TiDB 5.1 versions)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by overvenus** [comment]
@leoppro PTAL

**Review by amyangfei — APPROVED** [review]

File: ticdc/ticdc-overview.md [hunk 1]
```diff
@@ -123,3 +123,11 @@ TiCDC 从 4.0.8 版本开始，可通过修改任务配置来同步**没有有
 ## TiCDC 开放数据协议
 
 TiCDC Open Protocol 是一种行级别的数据变更通知协议，为监控、缓存、全文索引、分析引擎、异构数据库的主从复制等提供数据源。TiCDC 遵循 TiCDC Open Protocol，向 MQ (Message Queue) 等第三方数据媒介复制 TiDB 的数据变更。详细信息参考 [TiCDC 开放数据协议](/ticdc/ticdc-open-protocol.md)。
+
+## `sort-dir` 及 `data-dir` 配置项
+
+`sort-dir` 配置项给 TiCDC 内部的排序器指定临时文件目录，其作用在各版本有过一些不兼容更改：
```
Comment (TomShawn) [inline ticdc/ticdc-overview.md:129]:
```suggestion
`sort-dir` 配置项用于给 TiCDC 内部的排序器指定临时文件目录，其作用在各版本有过如下兼容性更改：
```

File: ticdc/ticdc-overview.md [hunk 2]
```diff
@@ -123,3 +123,11 @@ TiCDC 从 4.0.8 版本开始，可通过修改任务配置来同步**没有有
...
+
+- v4.0.11 及更小版本，以及 v5.0.0-rc 中，`sort-dir` 仅作为 changefeed 配置项，给 file sorter 和 unified sorter 指定临时文件目录。在这些版本中，file sorter 和 unified sorter 均未 GA，不推荐生产环境使用。如果有多个 changefeed 被配置使用了 `unified` 作为 `sort-engine`，那么实际使用的临时文件目录可能是任何一个 changefeed 的 `sort-dir` 配置，且每个 TiCDC 节点上使用的目录可能不一致。
```
Comment (TomShawn) [inline ticdc/ticdc-overview.md:131]:
```suggestion
- 在 v4.0.11 及之前的 v4.0 版本，以及在 v5.0.0-rc 中，`sort-dir` 仅作为 changefeed 配置项，给 file sorter 和 unified Sorter 指定临时文件目录。在这些版本中，file sorter 和 unified sorter **均不是**正式功能 (GA)，因此不推荐生产环境使用。

    如果有多个 changefeed 被配置使用了 `unified` 作为 `sort-engine`，那么实际使用的临时文件目录可能是任何一个 changefeed 的 `sort-dir` 配置，且每个 TiCDC 节点上使用的目录可能不一致。
```

File: ticdc/ticdc-overview.md [hunk 3]
```diff
@@ -123,3 +123,11 @@ TiCDC 从 4.0.8 版本开始，可通过修改任务配置来同步**没有有
...
+- v4.0.12, v4.0.13, v5.0.0 及 v5.0.1 中，`sort-dir` 可以作为 changefeed 配置项或 cdc server 配置项。但在默认情况下 changefeed 的 `sort-dir` 配置不会生效，而 cdc server 的 `sort-dir` 配置默认为 `/tmp/cdc_sort`. 建议生产环境仅配置 cdc server 的相关配置。如果用户使用 TiUP 部署 TiCDC，建议升级到最新 TiUP 版本并在 TiCDC server 配置中设置 `sorter.sort-dir` 一项。v4.0.13, v5.0.0 和 v5.0.1 中 unified sorter 是默认开启的，如果要升级到如上版本，请确保 TiCDC server 配置中的 `sorter.sort-dir` 已经被正确配置。
```
Comment (TomShawn) [inline ticdc/ticdc-overview.md:132]:
```suggestion
- v4.0.12、v4.0.13、v5.0.0 及 v5.0.1 中，`sort-dir` 可以作为 changefeed 配置项或 `cdc server` 配置项。但在默认情况下 changefeed 的 `sort-dir` 配置不会生效，而 `cdc server` 的 `sort-dir` 配置默认为 `/tmp/cdc_sort`。建议生产环境下仅配置 `cdc server` 的相关配置。

    如果你使用 TiUP 部署 TiCDC，建议升级到最新的 TiUP 版本并在 TiCDC server 配置中设置 `sorter.sort-dir` 一项。在 v4.0.13、v5.0.0 和 v5.0.1 中 unified sorter 是默认开启的，如果要将集群升级至这些版本，请确保 TiCDC server 配置中的 `sorter.sort-dir` 已经被正确配置。
```

File: ticdc/ticdc-overview.md [hunk 4]
```diff
@@ -123,3 +123,11 @@ TiCDC 从 4.0.8 版本开始，可通过修改任务配置来同步**没有有
...
+- v5.0.2 及 v5.1.0 中，`sort-dir` 被弃用，建议用户配置 `data-dir`. `data-dir` 可以通过最新版本的 TiUP 进行配置。这些版本中 unified sorter 是默认开启的，升级时请确保 `data-dir` 已经被正确配置，否则将默认使用 /tmp/cdc_data, 如果该目录所在设备空间不足，有可能出现硬盘空间不足的问题。之前配置的 changefeed 的 `sort-dir` 配置将会失效。
```
Comment (TomShawn) [inline ticdc/ticdc-overview.md:133]:
```suggestion
- v5.0.2 及 v5.1.0 中，`sort-dir` 被弃用，建议用户配置 `data-dir`。`data-dir` 可以通过最新版本的 TiUP 进行配置。这些版本中 unified sorter 是默认开启的，升级时请确保 `data-dir` 已经被正确配置，否则将默认使用 `/tmp/cdc_data`。如果该目录所在设备空间不足，有可能出现硬盘空间不足的问题。之前配置的 changefeed 的 `sort-dir` 配置将会失效。 
```

File: ticdc/ticdc-overview.md [hunk 5]
```diff
@@ -123,3 +123,13 @@ TiCDC 从 4.0.8 版本开始，可通过修改任务配置来同步**没有有
...
+
+## `sort-dir` 及 `data-dir` 配置项的兼容性说明
+
+`sort-dir` 配置项用于给 TiCDC 内部的排序器指定临时文件目录，其作用在各版本有过如下兼容性更改：
+
+|  版本  |  `sort-engine` 的使用  |  说明   |
+|  :---  |    :---               |  :--    |
+| v4.0.11 及之前的 v4.0 版本，v5.0.0-rc | 作为 changefeed 配置项，给 `file` sorter 和 `unified` Sorter 指定临时文件目录 | 在这些版本中，`file` sorter 和 `unified` sorter **均不是**正式功能 (GA)，不推荐在生产环境中使用。<br/><br/>如果有多个 changefeed 被配置使用了 `unified` 作为 `sort-engine`，那么实际使用的临时文件目录可能是任何一个 changefeed 的 `sort-dir` 配置，且每个 TiCDC 节点上使用的目录可能不一致。 |
+| v4.0.12，v4.0.13，v5.0.0 及 v5.0.1 | 作为 changefeed 配置项或 `cdc server` 配置项 | 在默认情况下 changefeed 的 `sort-dir` 配置不会生效，而 `cdc server` 的 `sort-dir` 配置默认为 `/tmp/cdc_sort`。建议生产环境下仅配置 `cdc server` 的相关配置。<br /><br />如果你使用 TiUP 部署 TiCDC，建议升级到最新的 TiUP 版本并在 TiCDC server 配置中设置 `sorter.sort-dir` 一项。<br /><br />在 v4.0.13、v5.0.0 和 v5.0.1 中 unified sorter 是默认开启的，如果要将集群升级至这些版本，请确保 TiCDC server 配置中的 `sorter.sort-dir` 已经被正确配置。|
+|  v5.0.2 及之后的 v5.0 版本，v5.1 | `sort-dir` 被弃用，建议配置 `data-dir` |  `data-dir` 可以通过最新版本的 TiUP 进行配置。这些版本中 unified sorter 是默认开启的，升级时请确保 `data-dir` 已经被正确配置，否则将默认使用 `/tmp/cdc_data`。<br /><br />如果该目录所在设备空间不足，有可能出现硬盘空间不足的问题。之前配置的 changefeed 的 `sort-dir` 配置将会失效。|
```
Comment (TomShawn) [inline ticdc/ticdc-overview.md:135]: PTAL "v5.0.2 及之后的 v5.0 版本，v5.1". Thanks. Shall we use "v5.0.2 及之后，以及其他更新的版本"？

Comment (amyangfei) [reply ticdc/ticdc-overview.md:135]: 可以的，相当于 >= 5.0.2

File: ticdc/ticdc-overview.md [hunk 6]
```diff
@@ -123,3 +123,13 @@ TiCDC 从 4.0.8 版本开始，可通过修改任务配置来同步**没有有
 ## TiCDC 开放数据协议
 
 TiCDC Open Protocol 是一种行级别的数据变更通知协议，为监控、缓存、全文索引、分析引擎、异构数据库的主从复制等提供数据源。TiCDC 遵循 TiCDC Open Protocol，向 MQ (Message Queue) 等第三方数据媒介复制 TiDB 的数据变更。详细信息参考 [TiCDC 开放数据协议](/ticdc/ticdc-open-protocol.md)。
+
+## `sort-dir` 及 `data-dir` 配置项的兼容性说明
+
+`sort-dir` 配置项用于给 TiCDC 内部的排序器指定临时文件目录，其作用在各版本有过如下兼容性更改：
+
+|  版本  |  `sort-engine` 的使用  |  说明   | 建议 |
```
Comment (TomShawn) [inline ticdc/ticdc-overview.md:131]:
```suggestion
|  版本  |  `sort-engine` 的使用  |  说明   |  使用建议  |
```

File: ticdc/ticdc-overview.md [hunk 7]
```diff
@@ -123,3 +123,13 @@ TiCDC 从 4.0.8 版本开始，可通过修改任务配置来同步**没有有
...
+|  :---  |    :---               |  :--    | :-- |
+| v4.0.11 及之前的 v4.0 版本，v5.0.0-rc | 作为 changefeed 配置项，给 `file` sorter 和 `unified` Sorter 指定临时文件目录 | 在这些版本中，`file` sorter 和 `unified` sorter **均不是**正式功能 (GA)，不推荐在生产环境中使用。<br/><br/>如果有多个 changefeed 被配置使用了 `unified` 作为 `sort-engine`，那么实际使用的临时文件目录可能是任何一个 changefeed 的 `sort-dir` 配置，且每个 TiCDC 节点上使用的目录可能不一致。 | 不推荐生产环境使用 Unified Sorter |
```
Comment (TomShawn) [inline ticdc/ticdc-overview.md:133]:
```suggestion
| v4.0.11 及之前的 v4.0 版本，v5.0.0-rc | 作为 changefeed 配置项，给 `file` sorter 和 `unified` Sorter 指定临时文件目录 | 在这些版本中，`file` sorter 和 `unified` sorter **均不是**正式功能 (GA)，不推荐在生产环境中使用。<br/><br/>如果有多个 changefeed 被配置使用了 `unified` 作为 `sort-engine`，那么实际使用的临时文件目录可能是任何一个 changefeed 的 `sort-dir` 配置，且每个 TiCDC 节点上使用的目录可能不一致。 | 不推荐生产环境中使用 Unified Sorter |
```

File: ticdc/ticdc-overview.md [hunk 8]
```diff
@@ -123,3 +123,13 @@ TiCDC 从 4.0.8 版本开始，可通过修改任务配置来同步**没有有
...
+| v4.0.12，v4.0.13，v5.0.0 及 v5.0.1 | 作为 changefeed 配置项或 `cdc server` 配置项 | 在默认情况下 changefeed 的 `sort-dir` 配置不会生效，而 `cdc server` 的 `sort-dir` 配置默认为 `/tmp/cdc_sort`。建议生产环境下仅配置 `cdc server` 的相关配置。<br /><br />如果你使用 TiUP 部署 TiCDC，建议升级到最新的 TiUP 版本并在 TiCDC server 配置中设置 `sorter.sort-dir` 一项。<br /><br />在 v4.0.13、v5.0.0 和 v5.0.1 中 unified sorter 是默认开启的，如果要将集群升级至这些版本，请确保 TiCDC server 配置中的 `sorter.sort-dir` 已经被正确配置。| 需要通过 cdc server 命令行参数 (或 TiUP) 配置 `sort-dir` |
```
Comment (TomShawn) [inline ticdc/ticdc-overview.md:134]:
```suggestion
| v4.0.12，v4.0.13，v5.0.0 及 v5.0.1 | 作为 changefeed 配置项或 `cdc server` 配置项 | 在默认情况下 changefeed 的 `sort-dir` 配置不会生效，而 `cdc server` 的 `sort-dir` 配置默认为 `/tmp/cdc_sort`。建议生产环境下仅配置 `cdc server` 的相关配置。<br /><br />如果你使用 TiUP 部署 TiCDC，建议升级到最新的 TiUP 版本并在 TiCDC server 配置中设置 `sorter.sort-dir` 一项。<br /><br />在 v4.0.13、v5.0.0 和 v5.0.1 中 unified sorter 是默认开启的，如果要将集群升级至这些版本，请确保 TiCDC server 配置中的 `sorter.sort-dir` 已经被正确配置。| 需要通过 `cdc server` 命令行参数（或 TiUP）配置 `sort-dir` |
```

File: ticdc/ticdc-overview.md [hunk 9]
```diff
@@ -123,3 +123,13 @@ TiCDC 从 4.0.8 版本开始，可通过修改任务配置来同步**没有有
...
+|  v5.0.2 及之后的 v5.0 版本，v5.1 | `sort-dir` 被弃用，建议配置 `data-dir` |  `data-dir` 可以通过最新版本的 TiUP 进行配置。这些版本中 unified sorter 是默认开启的，升级时请确保 `data-dir` 已经被正确配置，否则将默认使用 `/tmp/cdc_data`。<br /><br />如果该目录所在设备空间不足，有可能出现硬盘空间不足的问题。之前配置的 changefeed 的 `sort-dir` 配置将会失效。| 需要通过 cdc server 命令行参数 (或 TiUP)  配置 `data-dir` |
```
Comment (TomShawn) [inline ticdc/ticdc-overview.md:135]:
```suggestion
|  v5.0.2 及之后的 v5.0 版本，v5.1 | `sort-dir` 被弃用，建议配置 `data-dir` |  `data-dir` 可以通过最新版本的 TiUP 进行配置。这些版本中 unified sorter 是默认开启的，升级时请确保 `data-dir` 已经被正确配置，否则将默认使用 `/tmp/cdc_data`。<br /><br />如果该目录所在设备空间不足，有可能出现硬盘空间不足的问题。之前配置的 changefeed 的 `sort-dir` 配置将会失效。| 需要通过 `cdc server` 命令行参数（或 TiUP）配置 `data-dir` |
```

**Review by TomShawn — APPROVED** [review]

File: ticdc/ticdc-overview.md [hunk 10]
```diff
@@ -123,3 +123,13 @@ TiCDC 从 4.0.8 版本开始，可通过修改任务配置来同步**没有有
...
+
+|  版本  |  `sort-engine` 的使用  |  说明   |  使用建议  |
+|  :---  |    :---               |  :--    | :-- |
+| v4.0.11 及之前的 v4.0 版本，v5.0.0-rc | 作为 changefeed 配置项，给 `file` sorter 和 `unified` Sorter 指定临时文件目录 | 在这些版本中，`file` sorter 和 `unified` sorter **均不是**正式功能 (GA)，不推荐在生产环境中使用。<br/><br/>如果有多个 changefeed 被配置使用了 `unified` 作为 `sort-engine`，那么实际使用的临时文件目录可能是任何一个 changefeed 的 `sort-dir` 配置，且每个 TiCDC 节点上使用的目录可能不一致。 | 不推荐在生产环境中使用 Unified Sorter |
+| v4.0.12，v4.0.13，v5.0.0 及 v5.0.1 | 作为 changefeed 配置项或 `cdc server` 配置项 | 在默认情况下 changefeed 的 `sort-dir` 配置不会生效，而 `cdc server` 的 `sort-dir` 配置默认为 `/tmp/cdc_sort`。建议生产环境下仅配置 `cdc server` 的相关配置。<br /><br />如果你使用 TiUP 部署 TiCDC，建议升级到最新的 TiUP 版本并在 TiCDC server 配置中设置 `sorter.sort-dir` 一项。<br /><br />在 v4.0.13、v5.0.0 和 v5.0.1 中 unified sorter 是默认开启的，如果要将集群升级至这些版本，请确保 TiCDC server 配置中的 `sorter.sort-dir` 已经被正确配置。| 需要通过 `cdc server` 命令行参数（或 TiUP）配置 `sort-dir` |
+|  v5.0.2 及之后的 v5.0 版本，v5.1 | `sort-dir` 被弃用，建议配置 `data-dir` |  `data-dir` 可以通过最新版本的 TiUP 进行配置。这些版本中 unified sorter 是默认开启的，升级时请确保 `data-dir` 已经被正确配置，否则将默认使用 `/tmp/cdc_data`。<br /><br />如果该目录所在设备空间不足，有可能出现硬盘空间不足的问题。之前配置的 changefeed 的 `sort-dir` 配置将会失效。| 需要通过 `cdc server` 命令行参数（或 TiUP）配置 `data-dir` |
```
Comment (TomShawn) [inline ticdc/ticdc-overview.md:135]:
```suggestion
|  v5.0.2 及之后的版本  | `sort-dir` 被弃用，建议配置 `data-dir` |  `data-dir` 可以通过最新版本的 TiUP 进行配置。这些版本中 unified sorter 是默认开启的，升级时请确保 `data-dir` 已经被正确配置，否则将默认使用 `/tmp/cdc_data`。<br /><br />如果该目录所在设备空间不足，有可能出现硬盘空间不足的问题。之前配置的 changefeed 的 `sort-dir` 配置将会失效。| 需要通过 `cdc server` 命令行参数（或 TiUP）配置 `data-dir` |
```

File: ticdc/ticdc-overview.md [hunk 11]
```diff
@@ -123,3 +123,13 @@ TiCDC 从 4.0.8 版本开始，可通过修改任务配置来同步**没有有
...
+
+|  版本  |  `sort-engine` 的使用  |  说明   |  使用建议  |
+|  :---  |    :---               |  :--    | :-- |
+| v4.0.11 及之前的 v4.0 版本，v5.0.0-rc | 作为 changefeed 配置项，给 `file` sorter 和 `unified` Sorter 指定临时文件目录 | 在这些版本中，`file` sorter 和 `unified` sorter **均不是**正式功能 (GA)，不推荐在生产环境中使用。<br/><br/>如果有多个 changefeed 被配置使用了 `unified` 作为 `sort-engine`，那么实际使用的临时文件目录可能是任何一个 changefeed 的 `sort-dir` 配置，且每个 TiCDC 节点上使用的目录可能不一致。 | 不推荐在生产环境中使用 Unified Sorter |
+| v4.0.12，v4.0.13，v5.0.0 及 v5.0.1 | 作为 changefeed 配置项或 `cdc server` 配置项 | 在默认情况下 changefeed 的 `sort-dir` 配置不会生效，而 `cdc server` 的 `sort-dir` 配置默认为 `/tmp/cdc_sort`。建议生产环境下仅配置 `cdc server` 的相关配置。<br /><br />如果你使用 TiUP 部署 TiCDC，建议升级到最新的 TiUP 版本并在 TiCDC server 配置中设置 `sorter.sort-dir` 一项。<br /><br />在 v4.0.13、v5.0.0 和 v5.0.1 中 unified sorter 是默认开启的，如果要将集群升级至这些版本，请确保 TiCDC server 配置中的 `sorter.sort-dir` 已经被正确配置。| 需要通过 `cdc server` 命令行参数（或 TiUP）配置 `sort-dir` |
+|  v5.0.2 及之后的版本  | `sort-dir` 被弃用，建议配置 `data-dir` |  `data-dir` 可以通过最新版本的 TiUP 进行配置。这些版本中 unified sorter 是默认开启的，升级时请确保 `data-dir` 已经被正确配置，否则将默认使用 `/tmp/cdc_data`。<br /><br />如果该目录所在设备空间不足，有可能出现硬盘空间不足的问题。之前配置的 changefeed 的 `sort-dir` 配置将会失效。| 需要通过 `cdc server` 命令行参数（或 TiUP）配置 `data-dir` |
```
Comment (TomShawn) [inline ticdc/ticdc-overview.md:135]: Also include changes in https://github.com/pingcap/docs-cn/pull/6742/files#diff-723134ed55a23b9be44529ff13c37ea37026a393bd3bb8a28783d844676a99afR135.

**Pull request merged**